GDJS/Runtime/inputmanager.js

```diff
@@ -42,21 +42,65 @@ gdjs.InputManager.MOUSE_RIGHT_BUTTON = 1;
 gdjs.InputManager.MOUSE_MIDDLE_BUTTON = 2;
 
 /**
- * Should be called whenever a key is pressed
+ * Holds the keyCodes of the keys which only have left / right
+ * variants and should thus default to their left variant values
+ * if necessary.
+ *
+ * @constant {number}
+ */
+gdjs.InputManager.DEFAULT_LEFT_VARIANT_KEYS = [16, 17, 18, 91];
+
+/**
+ * Returns the location aware keyCode.
+ *
+ * @private
+ */
+gdjs.InputManager.prototype._getLocationAwareKeyCode = function(
+  keyCode,
+  location
+) {
+  if (location) {
+    // If it is a numpad number, do not modify it.
+    if (96 <= keyCode && keyCode <= 105) {
+      return keyCode;
+    }
+
+    return keyCode + 1000 * location;
+  }
+
+  if (gdjs.InputManager.DEFAULT_LEFT_VARIANT_KEYS.indexOf(keyCode) !== -1) {
+    return keyCode + 1000;
+  }
+
+  return keyCode;
+};
+
+/**
+ * Should be called whenever a key is pressed. The location corresponds to
+ * KeyboardEvent.location, which should be 0 for standard keys, 1 for left keys,
+ * 2 for right keys, and 3 for numpad keys.
  * @param {number} keyCode The key code associated to the key press.
+ * @param {number} location The location of the event.
  */
-gdjs.InputManager.prototype.onKeyPressed = function(keyCode) {
-  this._pressedKeys.put(keyCode, true);
-  this._lastPressedKey = keyCode;
+gdjs.InputManager.prototype.onKeyPressed = function(keyCode, location) {
+  var locationAwareKeyCode = this._getLocationAwareKeyCode(keyCode, location);
+
+  this._pressedKeys.put(locationAwareKeyCode, true);
+  this._lastPressedKey = locationAwareKeyCode;
 };
 
 /**
- * Should be called whenever a key is released
+ * Should be called whenever a key is released. The location corresponds to
+ * KeyboardEvent.location, which should be 0 for standard keys, 1 for left keys,
+ * 2 for right keys, and 3 for numpad keys.
  * @param {number} keyCode The key code associated to the key release.
+ * @param {number} location The location of the event.
```

Comment:
Use equal: `{number=}` to note an optional parameter 
(It's a JSDoc notation: https://jsdoc.app/tags-param.html#optional-parameters-and-default-values also understood by Typescript.)